orttraining/orttraining/python/ort_trainer.py

```diff
@@ -713,6 +714,19 @@ def prepare_input_and_fetches(self, input_desc_with_, internal_learning_rate, in
         if internal_loss_scale is not None:
             input = input + (internal_loss_scale,)
 
+        # loss_scale input name is needed to call train_step.
+        # loss_scale input name is determined at the time training session is created.
+        # training session creation is delay to train_step.
+        # as a result, loss_scale input name is not available when calling train_step. 
+        # to work around this problem, we have to use the following pattern matching code to get the loss_scaler argument.
+        # this problem can be solved if leter it is decided that loss scaler is always internal.
+        extra_kargs_keys = [key for key in kwargs.keys() if key not in [desc.name_ for desc in input_desc_with_]]
+        if len(input_desc_with_) > len(input) and len(extra_kargs_keys) == 1 and internal_loss_scale is None:
```

Comment:
I think the issue can be resolved by another way. The loss scale is only needed after the exported onnx model is processed by ORTTrainer C++ backend, and by that time the session object is created. 

So when exporting onnx model for the first time, we can work without loss scale input. After export is done, and session is created, we have loss scale input name, and everything continues as normal.

Reply:
But loss scale is needed at train_step where session is not create.